sycl/source/detail/scheduler/commands.cpp

```diff
@@ -585,14 +596,14 @@ const ContextImplPtr &Command::getWorkerContext() const {
   return MQueue->getContextImplPtr();
 }
 
-const QueueImplPtr &Command::getWorkerQueue() const { return MQueue; }
+const QueueImplPtr &Command::getWorkerQueue() const {
+  assert(MWorkerQueue && "MWorkerQueue must not be nullptr");
+  return MWorkerQueue;
+}
 
 bool Command::producesPiEvent() const { return true; }
 
-bool Command::supportsPostEnqueueCleanup() const {
-  // Isolated commands are cleaned up separately
-  return !MUsers.empty() || !MDeps.empty();
-}
+bool Command::supportsPostEnqueueCleanup() const { return true; }
```

Comment:
This function is not virtual, and it's not doing anything useful. Why do we need it?

Reply:
/// Returns true iff this command can be freed by post enqueue cleanup.
  virtual bool supportsPostEnqueueCleanup() const;

it is virtual, and some accessors of Command overrides it (e.g. ExecCGCommand)

Reply:
I see, ok